src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -256,16 +371,42 @@ export class DataResourceDataProvider implements IGridDataProvider {
 		});
 	}
 
+	public async convertAllData(result: ResultSetSummary): Promise<void> {
+		// Querying 50 rows at a time. Querying large amount of rows will be slow and
```

Comment:
Have you done any benchmarking here? Would love to understand any perf differences between 50, 25, and 100 here (for example), on a table with 10k rows.

Reply:
There weren't any noticeable differences for 10k rows but when I tested on 500k rows, querying 100 rows at a time made the conversion process faster by a couple of seconds (~4 seconds) and didn't affect the table rendering. I can change the query amount to 100.

Reply:
I also tested larger query amounts (500 rows) but sometimes it would slow down the table rendering.